packages/jsii-reflect/lib/source.ts

```diff
@@ -0,0 +1,49 @@
+import { Assembly } from "./assembly";
+
+export interface SourceLocation {
+  filename: string;
+  line: number;
+}
+
+interface Locatable {
+  readonly assembly: Assembly;
+  readonly moduleLocation?: SourceLocation;
+}
+
+/**
+ * Return the repository location for the given API item
+ */
+export function repositoryLocation(item: Locatable): SourceLocation | undefined {
+  const assemblyLoc = item.assembly.repositoryLocation;
+  const moduleLoc = item.moduleLocation;
+
+  if (assemblyLoc && moduleLoc) {
+    if (assemblyLoc === '.') { return moduleLoc; }
+
+    return {
+      filename: `${assemblyLoc}/${moduleLoc.filename}`,
+      line: moduleLoc.line,
+    };
+  }
+
+  return undefined;
+}
+
+/**
+ * Return a URL for this item into the source repository, if available
+ *
+ * (Currently only supports GitHub URLs)
+ */
+export function repositoryUrl(item: Locatable, ref: string = 'master'): string | undefined {
+  const loc = repositoryLocation(item);
+  if (!loc) { return undefined; }
+
+  const repo = item.assembly.repository;
+  if (!repo.url.startsWith('https://') || repo.url.indexOf("github.com") === -1 || !repo.url.endsWith('.git')) { return undefined; }
+
+  // Turn https://github.com/awslabs/aws-cdk.git ->  https://github.com/awslabs/aws-cdk/blob/REF/filename#L<number>
```

Comment:
Would be interesting to look into making this commit or tag-specific, so you get to the exact version of the file that corresponds to the doc you're looking at.

Reply:
Yes, would be nice. I suppose the only solution for that is embedding git commits in the assembly as well, but then the question becomes how we're going to get those in there.

Reply:
I guess code generation could provide context (e.g: git commit-ish, which is a fancy way to mean commit SHA, tag, branch, ...) that is then passed there & used? Can be done later, though (maybe create issue).